src/clientiface.cpp

```diff
@@ -379,19 +381,10 @@ void RemoteClient::GetNextBlocks (
 				continue;
 			}
 
-			/*
-				If block has been marked to not exist on disk (dummy) or is
-				not generated and generating new ones is not wanted, skip block.
-			*/
-			if (!generate && block_not_found) {
-				// get next one.
-				continue;
-			}
-
 			/*
 				Add inexistent block to emerge queue.
 			*/
-			if (block == NULL || block_not_found) {
+			if (block == NULL || !block->isGenerated()) {
```

Comment:
```suggestion
			if (!block || !block->isGenerated()) {
```

Reply:
Hah. I had this first, but then did not want to change it. I'll make this change.